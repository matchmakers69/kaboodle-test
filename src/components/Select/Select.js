import React from 'react';
import PropTypes from 'prop-types';

const Select = ({
  value,
  name,
  onChange,
  options,
  'data-test': dataTest,
  placeholder
}) => {
  return (
    <select value={value} name={name} data-test={dataTest} onChange={onChange}>
      {placeholder && (
        <option value='' disabled>
          {placeholder}
        </option>
      )}
      {options.map(opt => (
        <option key={opt.label} value={opt.value} label={opt.label}>
          {opt.label}
        </option>
      ))}
    </select>
  );
};

Select.propTypes = {
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  'data-test': PropTypes.string,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.array.isRequired,
  placeholder: PropTypes.string.isRequired
};

export default Select;
