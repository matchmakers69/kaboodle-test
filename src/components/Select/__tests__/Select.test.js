import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Select from '../Select';

configure({ adapter: new Adapter() });

const props = {
  onChange: jest.fn(),
  placeholder: 'Choose rating',
  value: '',
  type: '',
  name: '',
  options: []
};

describe('<Select />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Select {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
