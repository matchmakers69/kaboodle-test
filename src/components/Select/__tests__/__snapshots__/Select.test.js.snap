// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders correctly 1`] = `
<select
  name=""
  onChange={[MockFunction]}
  value=""
>
  <option
    disabled={true}
    value=""
  >
    Choose rating
  </option>
</select>
`;
